src/compiler/checker.ts

```diff
@@ -6204,7 +6204,7 @@ namespace ts {
                             && isTypeIdenticalTo(getTypeOfSymbol(p), getTypeOfPropertyOfType(baseType, p.escapedName)!))) {
                             return [];
                         }
-                        const flag = modifierFlags | (isStatic ? ModifierFlags.Static : 0);
+                        const flag = (modifierFlags & ModifierFlags.Async ? modifierFlags & ~ModifierFlags.Async : modifierFlags) | (isStatic ? ModifierFlags.Static : 0);

```

Comment:
```suggestion
                        const flag = (modifierFlags & ~ModifierFlags.Async) | (isStatic ? ModifierFlags.Static : 0);

```

you don't need to bother with the check~

Reply:
@weswigham Oke. Thanks!